### DISCUSSION TITLE
self.pixel_per_meter  parameter in Distance calculation

### DISCUSSION THREAD

**Discussion by mr-mainak** [ultralytics/ultralytics#8701] [General]
Hi, greetings of the day.
Can anybody please give reference to how to find pixel_per_meter? In here I see it's set to 10. How to find this?
Regards
@glenn-jocher

**Comment by glenn-jocher** [comment]
Hello! 👋

The `pixel_per_meter` parameter is essentially a scaling factor that converts measurements from pixels to meters in your specific application. Its value depends on the scale of objects in your images and the distance from the camera to the objects.

To find `pixel_per_meter`, you need to know the actual size of an object in the image (in meters) and its size in pixels. For example, if you know a 2-meter tall object appears as 200 pixels in height in your image, then `pixel_per_meter = 200 pixels / 2 meters = 100`.

Here's a quick example:

```python
actual_height_m = 2  # height in meters
height_in_pixels = 200  # height in pixels
pixel_per_meter = height_in_pixels / actual_height_m
```

Adjust this parameter based on your specific scenario for accurate distance calculations. Hope this helps!

**Reply by mr-mainak** [reply]
@glenn-jocher Thank you

**Reply by glenn-jocher** [reply]
@mr-mainak hello! 😊

To find the `pixel_per_meter` value, you'll need a reference object in your image whose real-world size you know. For example, if a 2m tall object spans 200 pixels in your image, then `pixel_per_meter = 200 / 2 = 100`. Adjust this based on your specific setup for accurate measurements.

```python
# Example calculation
real_world_size_m = 2  # meters
object_size_pixels = 200  # pixels
pixel_per_meter = object_size_pixels / real_world_size_m
```

Hope this helps! Let us know if you have any more questions.
